flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/TemporalRowtimeJoin.scala

```diff
@@ -272,24 +264,30 @@ class TemporalRowtimeJoin(
     * Binary search `rightRowsSorted` to find the latest right row to join with `leftTime`.
     * Latest means a right row with largest time that is still smaller or equal to `leftTime`.
     *
-    * @return index of such element. If such row was not found (either `rightRowsSorted` is empty
-    *         or all `rightRowsSorted` are are newer) return -1.
+    * @return found element or `Option.empty` If such row was not found (either `rightRowsSorted`
+    *         is empty or all `rightRowsSorted` are are newer).
     */
-  private def latestRightRowToJoin(rightRowsSorted: util.List[Row], leftTime: Long): Int = {
+  private def latestRightRowToJoin(rightRowsSorted: util.List[Row], leftTime: Long): Option[Row] = {
```

Comment:
nit: use `null` instead of option

Reply:
`Option` > `NullPointerException` :( Do not use nulls without working `@Nullable` annotations